c++/src/TypeImpl.cc

```diff
@@ -413,6 +413,8 @@ namespace orc {
   std::unique_ptr<Type> convertType(const proto::Type& type,
                                     const proto::Footer& footer) {
     std::unique_ptr<Type> ret;
+    //check for corrupt footers before conversion
+    checkProtoTypes(footer);
```

Comment:
Could you describe what happens after this PR? Previously, you filed an issue with `it crashed`.

Reply:
> Could you describe what happens after this PR? Previously, you filed the issue with `it crashed`.

@dongjoon-hyun , Thanks for reviewing the PR. I can update as below : 

// Validate the indices in the Protobuf message type tree especially for the Type_Kind_STRUCT when fieldnames are less than the subtypes_size() , so we won't crash when we convert the proto::Types to TypeImpls.

Reply:
Please put it into the PR description instead of commenting here.
The PR description will be a commit message later.

Reply:
Done.